### PR TITLE
Fix: Adding and Removing public config take long time to reflect

### PR DESCRIPTION
**Pull request by Mahmoud-Emad**
### Description

Refactored and enhanced the AddPublicConfig component to improve readability and maintainability. Extracted configuration fields and used a loop to generate input-validator components for each field. Also, improved variable naming and removed redundant comments.

### Changes

- Created a new variable called `defaultNodeConfig` and copied its values to the config. The form now watches these values instead of `new` and `old` values.
- Extracted configuration fields to improve code readability.
- Improved consistency in variable naming.
- Removed redundant comments and added explanatory comments where needed.
- Created a fake data with `Fill Fake Data` button on development mode.

### Related Issues
- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/1855

### Screenshots/Video

[Screencast from 01-30-2024 05:57:31 PM.webm](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/57001890/7750ea8d-6e7b-49c8-be1f-26d1f2e9442d)

### Checklist
- [ ] Tests included
- [x] Build pass
- [x] Documentation
- [x] Code format and docstrings
- [x] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

File: packages/playground/src/dashboard/components/public_config.vue [hunk 1]
```diff
@@ -77,30 +82,39 @@
...
+              @click="isNodeHasConfig"
               color="error"
               variant="outlined"
               :disabled="isRemoving || Object.values(config).every(value => value == '')"
-              >Remove Config</v-btn
             >
+              Remove Config
+            </v-btn>
+
+            <v-btn v-if="runningInDevMode" @click="fillFakeData" color="warning" variant="outlined">
```
Comment (AlaaElattar) [inline packages/playground/src/dashboard/components/public_config.vue:113]: do we need this btn ?

Comment (Mahmoud-Emad) [reply packages/playground/src/dashboard/components/public_config.vue:113]: Yes it helps us in development mode to fill fake data, don't worry it won't appear in the production, only development

**Review by mohamedamer453 — APPROVED** [review]
However i'm not sure about the `Fill Fake Data` button, imo it should be removed. even if it only appears in dev environment. developing with the same data every time might lead to unwanted behaviors.

**Comment by Mahmoud-Emad** [comment]
> developing with the same data every time might lead to unwanted behaviors.

Yeah, you have a point, I removed it, kindly take a look again :)

**Review by MohamedElmdary — APPROVED** [review]

**Pull request merged**